pkg/apis/v1/nodeclaim_conversion_test.go

```diff
@@ -0,0 +1,539 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1_test
+
+import (
+	"github.com/awslabs/operatorpkg/status"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/samber/lo"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+
+	. "sigs.k8s.io/karpenter/pkg/apis/v1"
+	"sigs.k8s.io/karpenter/pkg/apis/v1beta1"
+	"sigs.k8s.io/karpenter/pkg/operator/injection"
+	"sigs.k8s.io/karpenter/pkg/test"
+)
+
+var (
+	v1NodeClaim      *NodeClaim
+	v1beta1NodeClaim *v1beta1.NodeClaim
+)
+
+var _ = Describe("Convert v1 to v1beta1 NodeClaim API", func() {
+	BeforeEach(func() {
+		v1beta1NodePool = test.NodePool()
+		v1NodePool = &NodePool{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "test-nodepool",
+			},
+			Spec: NodePoolSpec{
+				Template: NodeClaimTemplate{
+					Spec: NodeClaimSpec{
+						NodeClassRef: &NodeClassReference{
+							Name:  "test",
+							Kind:  "test-kind",
+							Group: "test-group",
+						},
+						Requirements: []NodeSelectorRequirementWithMinValues{},
+					},
+				},
+			},
+		}
+		v1NodeClaim = &NodeClaim{}
```

Comment:
nit: It's not standard convention to use capitalization like this in golang. Should we lowercase our variables throughout this file?